settings-portal/Settings.vala

```diff
@@ -0,0 +1,155 @@
+/*-
+ * Copyright 2021 elementary, Inc. (https://elementary.io)
+ * Copyright 2021 Alexander Mikhaylenko <alexm@gnome.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+ * Boston, MA 02110-1335 USA.
+ */
+
+[DBus (name = "org.freedesktop.portal.Error")]
+public errordomain PortalError {
+    FAILED,
+    INVALID_ARGUMENT,
+    NOT_FOUND,
+    EXISTS,
+    NOT_ALLOWED,
+    CANCELLED,
+    WINDOW_DESTROYED
+}
+
+[DBus (name = "io.elementary.pantheon.AccountsService")]
+private interface Pantheon.AccountsService : Object {
+    public abstract int prefers_color_scheme { owned get; set; }
+}
+
+[DBus (name = "org.freedesktop.Accounts")]
+interface FDO.Accounts : Object {
+    public abstract string find_user_by_name (string username) throws GLib.Error;
+}
+
+/* Copied from Granite.Settings */
+private class AccountsServiceMonitor : GLib.Object {
+    private FDO.Accounts? accounts_service = null;
+    private Pantheon.AccountsService? pantheon_act = null;
+    private string user_path;
+
+    public int32 color_scheme { get; set; }
```

Comment:
I forgot this gets exported. It shouldn't be.

Reply:
I don't think so, The AccountsServiceMonitor isn't declared as a DBus interface so this definitively don't get exported.

Reply:
`AccountsServiceMonitor`? I'm pointing at `color_scheme` which is a public prop.

Reply:
It's a public property of the `AccountsServiceMonitor`, only the `Settings` class and they public properties get exported.